### PR TITLE
 Adds validation for max dosage in PRN prescription #8324

### LINKED ISSUE #8324: Add validation for max dosage in PRN prescription
image

Add frontend and backend validation to check that max dosage in 24 is greater than or equal to base dosage

backend issue: https://github.com/coronasafe/care/issues/2383

### PR DESCRIPTION
**Pull request by Nithin9585**
## Proposed Changes

- Fixes #8324
- Added validation for maximum dosage in the `validators.ts` file.
- Updated UI to show dosage warnings when limits are exceeded.

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate bug / test a new feature.
- [x] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [] Request for Peer Reviews
- [] Completion of QA

Screenshot 2024-08-21 000732

### CODE REVIEW

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

File: src/Components/Medicine/validators.ts [hunk 1]
```diff
@@ -6,27 +6,51 @@ import { Prescription } from "./models";
...
+
       if (
         form.base_dosage &&
         form.target_dosage &&
         form.base_dosage.split(" ")[1] !== form.target_dosage.split(" ")[1]
       ) {
-        errors.base_dosage = "Unit must be same as target dosage's unit";
-        errors.target_dosage = "Unit must be same as base dosage's unit";
+        errors.base_dosage = "Unit must be the same as target dosage's unit";
+        errors.target_dosage = "Unit must be the same as base dosage's unit";
```
Comment (khavinshankar) [inline src/Components/Medicine/validators.ts]: use localization for error messages

**Review by khavinshankar — DISMISSED** [review]
@Nithin9585 good work, just i18n the error strings

**Review by bodhish — APPROVED** [review]
LGTM

**Review by gigincg — APPROVED** [review]

**Pull request merged**